frontend/src/types/global.d.ts

```diff
@@ -0,0 +1,387 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+// with the License. A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0/
+//
+// or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+    ComputeFleetStatus,
+    InstanceState,
+    NodeType,
+    ImageBuildStatus,
+    ImageBuilderImageStatus,
+    Ec2AmiState,
+    ClusterStatus,
+    ValidationLevel,
+    CloudFormationStackStatus,
+    CloudFormationResourceStatus,
+    JobStateCode,
+} from './enums'
+
+
+export {}
+
+declare global {
```

Comment:
issue(non-blocking): Global types are ok, yet I would rather split this big file into smaller files according to their domain (e.g Instance, Image, Jobs, Logs, and so on)

The reasoning behind it would be to have a clearer separation between what part of the codebase has ownership over what types.

To give a maybe oversimplified example: if we got rid of the whole `Images` section, we could forget to remove the related types.

Thoughts on this?

Reply:
Some of the types are common, but that can also go in its own separate file.

I wasn't able to figure out a good way to co-locate the constants, so in some ways those would also need to reflect the same structure -- which might make it more inconvenient from the consumption side.

I'm not sure if we can / should get rid of something like the Images section -- since this is a copy of the types in PC I would expect it to only expand for the life of PC3.x -- though perhaps they should be put into some PC specific and then Slurm specific?

I'm open to whatever, though might do that re-org as a separate pass (feel free to assign the work to me)

Reply:
Regarding the global types I would keep things consistent: there are some places where you import constants directly (`import { ClusterStatus } from '../../types/constants'`) and some places where types get included automatically. I would pick the same method (I prefer importing modules since it's more explicit and easier to track down but no strong preference here)

Regarding the splitting: I would keep an auto-generated-like file **only** if we can run a command (for example if PC releases a new version) and get the updated types with no manual intervention, so we can "forget" about types. Otherwise, it will be better to do even a rough separation between those types, like:
- `base` types
- `jobs`
- `cluster`
- `instances`
- `images`
- and so on

If you choose the first approach (keep everything in the same file) I encourage you to write that part and the related commands inside the readme/package.json, so a developer could use it in the future.